src/adagrams.js

```diff
@@ -1,15 +1,94 @@
+import {
+  LETTER_POOL,
+  SCORE_DICTIONARY
+} from "constants";
+
+const randomChoice = array => {
+  return array[Math.floor(array.length * Math.random())];
+};
+
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  let hand = [];
+  let listOfLetters = Object.keys(LETTER_POOL);
+  let copyOfLetterPool = { ...LETTER_POOL };
+
+  while (hand.length < 10) {
+    let letter = randomChoice(listOfLetters);
+
+    if (copyOfLetterPool[letter] === 0) {
+      continue;
+    } else {
+      copyOfLetterPool[letter] --;
+      hand.push(letter);
+    };
+  };
+  return hand;
 };
 
+
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  let lettersInHandDict = {};
+
+  lettersInHand.forEach(function(letter) {
+    if (letter in lettersInHandDict) {
+      lettersInHandDict[letter] ++;
+    } else {
+      lettersInHandDict[letter] = 1;
+    };
+  });
+
+  for (let character of input) {
+    if (character in lettersInHandDict && lettersInHandDict[character] > 0) {
+      lettersInHandDict[character] --;
+      } else {
+        return false;
+      };
+    };
+
+  return true;
 };
 
 export const scoreWord = (word) => {
-  // Implement this method for wave 3
+  let convertedWord = word.toUpperCase()
+  let wordScore = 0;
+
+  if (convertedWord.length < 1) {
+    return wordScore;
+  };
+
+  for (let letter of convertedWord) {
```

Comment:
this could also be const letter of convertedWord because its not being reassigned